### PR TITLE
Add directory for snippet docs and include README.md

### PR DESCRIPTION
**Pull request by nerrad**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woodocs/issues/125

This PR creates a directory for housing Code Snippets docs that will get migrated as a part of our documentation revamp. I've left the README.md pretty basic for now until we can determine what might get added to it. 

Merging this will allow code-snippets to be added in parallel PRs.

### CODE REVIEW

File: docs/snippets/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# WooCommerce Code Snippets
```
Comment (nielslange) [inline docs/snippets/README.md]:
```suggestion
# WooCommerce code snippets
```

We should use sentence case for the title, as mentioned on https://github.com/woocommerce/woodocs/blob/0d037666cbf2fc8056c89a1b67de3b7fe08c895a/style-guide.md?plain=1#L16-L17

**Review by nielslange — CHANGES_REQUESTED** [review]
Thanks for working on this, @nerrad. I left a suggestion regarding using sentence case, as mentioned in our Woo docs style guide.

**Comment by nerrad** [comment]
Good catch Niels and thanks for looking for that. Ideally, we'd get the MD linting into the Woo Core repo so style guide enforcement can be automated. 

Regarding this PR and #39177, would you mind taking care of merging them once you've done another review pass?

**Comment by nielslange** [comment]
> Good catch Niels and thanks for looking for that. Ideally, we'd get the MD linting into the Woo Core repo so style guide enforcement can be automated.

Great suggestion, @nerrad. I just created #39202 for tracking this.

> Regarding this PR and #39177, would you mind taking care of merging them once you've done another review pass?

Sure. I'll take care of both PRs.

**Review by nielslange — APPROVED** [review]
LGTM! 🙌

**Pull request merged**